odoo/models.py

```diff
@@ -69,8 +68,15 @@
 import typing
 if typing.TYPE_CHECKING:
     from collections.abc import Reversible
+    from .api import Environment, IdType, Self, ValuesType
     from .modules.registry import Registry
-    from odoo.api import Self, ValuesType, IdType
+    from .tools.translate import LazyGettext
+
+    ConstraintMessageType = (
+        str
+        | LazyGettext
```

Comment:
You shouldn't accept it normally? The translation is managed differently right ?